Build failure with PG11: logutriga.c:46:24: error: invalid type argument of '->'

pgq fails to build against PG 11beta1:
```
22:05:53 make[3]: Entering directory '/<<PKGBUILDDIR>>/triggers'
22:05:53 gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -g -g -O2 -fdebug-prefix-map=/build/postgresql-11-LcCi7f/postgresql-11-11~beta1=. -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -I. -I./ -I/usr/include/postgresql/11/server -I/usr/include/postgresql/internal  -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o logtriga.o logtriga.c
22:05:53 gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -g -g -O2 -fdebug-prefix-map=/build/postgresql-11-LcCi7f/postgresql-11-11~beta1=. -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -I. -I./ -I/usr/include/postgresql/11/server -I/usr/include/postgresql/internal  -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o logutriga.o logutriga.c
22:05:54 logutriga.c: In function 'pgq_urlenc_row':
22:05:54 logutriga.c:46:24: error: invalid type argument of '->' (have 'FormData_pg_attribute {aka struct FormData_pg_attribute}')
22:05:54    if (tupdesc->attrs[i]->attisdropped)
22:05:54                         ^~
22:05:54 make[3]: *** [<builtin>: logutriga.o] Error 1
```

--- a/triggers/common.c
+++ b/triggers/common.c
@@ -127,9 +127,9 @@ static void fill_magic_columns(PgqTriggerEvent *ev)
 
 	for (i = 0; i < tupdesc->natts; i++) {
 		/* Skip dropped columns */
-		if (tupdesc->attrs[i]->attisdropped)
+		if (TupleDescAttr(tupdesc, i)->attisdropped)
 			continue;
-		col_name = NameStr(tupdesc->attrs[i]->attname);
+		col_name = NameStr(TupleDescAttr(tupdesc, i)->attname);
 		if (!is_magic_field(col_name))
 			continue;
 		if (strcmp(col_name, "_pgq_ev_type") == 0)
@@ -224,9 +224,14 @@ static void init_cache(void)
 	 */
 	tbl_cache_ctx = AllocSetContextCreate(TopMemoryContext,
 					      "pgq_triggers table info",
+#if (PG_VERSION_NUM >= 110000)
+					      ALLOCSET_SMALL_SIZES
+#else
 					      ALLOCSET_SMALL_MINSIZE,
 					      ALLOCSET_SMALL_INITSIZE,
-					      ALLOCSET_SMALL_MAXSIZE);
+					      ALLOCSET_SMALL_MAXSIZE
+#endif
+					      );
 
 	/*
 	 * init pkey cache.
@@ -502,7 +507,7 @@ static void parse_oldstyle_args(PgqTriggerEvent *ev, TriggerData *tg)
 	 */
 	tupdesc = tg->tg_relation->rd_att;
 	for (i = 0, attcnt = 0; i < tupdesc->natts; i++) {
-		if (!tupdesc->attrs[i]->attisdropped)
+		if (!TupleDescAttr(tupdesc, i)->attisdropped)
 			attcnt++;
 	}
 
@@ -624,9 +629,9 @@ bool pgqtriga_skip_col(PgqTriggerEvent *ev, int i, int attkind_idx)
 	const char *name;
 
 	tupdesc = tg->tg_relation->rd_att;
-	if (tupdesc->attrs[i]->attisdropped)
+	if (TupleDescAttr(tupdesc, i)->attisdropped)
 		return true;
-	name = NameStr(tupdesc->attrs[i]->attname);
+	name = NameStr(TupleDescAttr(tupdesc, i)->attname);
 
 	if (is_magic_field(name)) {
 		ev->tgargs->custom_fields = 1;
@@ -658,9 +663,9 @@ bool pgqtriga_is_pkey(PgqTriggerEvent *ev, int i, int attkind_idx)
 		return ev->attkind[attkind_idx] == 'k';
 	} else if (ev->pkey_list) {
 		tupdesc = tg->tg_relation->rd_att;
-		if (tupdesc->attrs[i]->attisdropped)
+		if (TupleDescAttr(tupdesc, i)->attisdropped)
 			return false;
-		name = NameStr(tupdesc->attrs[i]->attname);
+		name = NameStr(TupleDescAttr(tupdesc, i)->attname);
 		if (is_magic_field(name)) {
 			ev->tgargs->custom_fields = 1;
 			return false;
@@ -852,7 +857,7 @@ int pgq_is_interesting_change(PgqTriggerEvent *ev, TriggerData *tg)
 		/*
 		 * Ignore dropped columns
 		 */
-		if (tupdesc->attrs[i]->attisdropped)
+		if (TupleDescAttr(tupdesc, i)->attisdropped)
 			continue;
 		attkind_idx++;
 

--- a/triggers/common.h
+++ b/triggers/common.h
@@ -1,3 +1,8 @@
+#if PG_VERSION_NUM < 100000
+/* from src/include/access/tupdesc.h, introduced in 2cd708452 */
+#define TupleDescAttr(tupdesc, i) ((tupdesc)->attrs[(i)])
+#endif
+
 enum PgqFields {
 	EV_TYPE = 0,
 	EV_DATA,

--- a/triggers/jsontriga.c
+++ b/triggers/jsontriga.c
@@ -164,7 +164,7 @@ static void pgq_jsonenc_row(PgqTriggerEvent *ev, HeapTuple row, StringInfo buf)
 	appendStringInfoChar(buf, '{');
 	for (i = 0; i < tg->tg_relation->rd_att->natts; i++) {
 		/* Skip dropped columns */
-		if (tupdesc->attrs[i]->attisdropped)
+		if (TupleDescAttr(tupdesc, i)->attisdropped)
 			continue;
 
 		attkind_idx++;
@@ -183,7 +183,7 @@ static void pgq_jsonenc_row(PgqTriggerEvent *ev, HeapTuple row, StringInfo buf)
 		appendStringInfoChar(buf, ':');
 
 		/* quote column value */
-		col_type = tupdesc->attrs[i]->atttypid;
+		col_type = TupleDescAttr(tupdesc, i)->atttypid;
 		col_datum = SPI_getbinval(row, tupdesc, i + 1, &isnull);
 		col_value = NULL;
 		if (isnull) {

--- a/triggers/logutriga.c
+++ b/triggers/logutriga.c
@@ -43,7 +43,7 @@ void pgq_urlenc_row(PgqTriggerEvent *ev, HeapTuple row, StringInfo buf)
 
 	for (i = 0; i < tg->tg_relation->rd_att->natts; i++) {
 		/* Skip dropped columns */
-		if (tupdesc->attrs[i]->attisdropped)
+		if (TupleDescAttr(tupdesc, i)->attisdropped)
 			continue;
 
 		attkind_idx++;

--- a/triggers/makesql.c
+++ b/triggers/makesql.c
@@ -71,7 +71,7 @@ static void process_insert(PgqTriggerEvent *ev, StringInfo sql)
 		char *col_ident;
 
 		/* Skip dropped columns */
-		if (tupdesc->attrs[i]->attisdropped)
+		if (TupleDescAttr(tupdesc, i)->attisdropped)
 			continue;
 
 		/* Check if allowed by colstring */
@@ -103,7 +103,7 @@ static void process_insert(PgqTriggerEvent *ev, StringInfo sql)
 		char *col_value;
 
 		/* Skip dropped columns */
-		if (tupdesc->attrs[i]->attisdropped)
+		if (TupleDescAttr(tupdesc, i)->attisdropped)
 			continue;
 
 		/* Check if allowed by colstring */
@@ -154,7 +154,7 @@ static int process_update(PgqTriggerEvent *ev, StringInfo sql)
 		/*
 		 * Ignore dropped columns
 		 */
-		if (tupdesc->attrs[i]->attisdropped)
+		if (TupleDescAttr(tupdesc, i)->attisdropped)
 			continue;
 
 		attkind_idx++;
@@ -240,7 +240,7 @@ static int process_update(PgqTriggerEvent *ev, StringInfo sql)
 			return 0;
 
 		for (i = 0, attkind_idx = -1; i < tupdesc->natts; i++) {
-			if (tupdesc->attrs[i]->attisdropped)
+			if (TupleDescAttr(tupdesc, i)->attisdropped)
 				continue;
 
 			attkind_idx++;
@@ -259,7 +259,7 @@ static int process_update(PgqTriggerEvent *ev, StringInfo sql)
 		/*
 		 * Ignore dropped columns
 		 */
-		if (tupdesc->attrs[i]->attisdropped)
+		if (TupleDescAttr(tupdesc, i)->attisdropped)
 			continue;
 
 		attkind_idx++;
@@ -291,7 +291,7 @@ static void process_delete(PgqTriggerEvent *ev, StringInfo sql)
 	int attkind_idx;
 
 	for (i = 0, attkind_idx = -1; i < tupdesc->natts; i++) {
-		if (tupdesc->attrs[i]->attisdropped)
+		if (TupleDescAttr(tupdesc, i)->attisdropped)
 			continue;
 
 		attkind_idx++;
@@ -323,7 +323,7 @@ int pgqtriga_make_sql(PgqTriggerEvent *ev, StringInfo sql)
 	 * Count number of active columns
 	 */
 	for (i = 0, attcnt = 0; i < tupdesc->natts; i++) {
-		if (tupdesc->attrs[i]->attisdropped)
+		if (TupleDescAttr(tupdesc, i)->attisdropped)
 			continue;
 		attcnt++;
 	}

